core/src/main/scala/kafka/message/ByteBufferMessageSet.scala

```diff
@@ -120,22 +205,87 @@ object ByteBufferMessageSet {
  *
  * Option 2: Give it a list of messages along with instructions relating to serialization format. Producers will use this method.
  *
+ *
+ * When message format v1 is used, there will be following message format changes.
+ * - For non-compressed messages, with message v1 we are adding timestamp and timestamp type attribute. The offsets of
+ *   the messages remain absolute offsets.
+ * - For Compressed messages, with message v1 we are adding timestamp, timestamp type attribute bit and using relative
+ *   offsets for inner messages of compressed messages. Timestamp type attribute is only set in wrapper messages.
```

Comment:
Since the inner offset is not really the relative offset, we can probably just refer to it as inner offset and point to the description in the later text.